akka-http/src/main/scala/akka/http/scaladsl/server/HttpApp.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.server
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.stream.ActorMaterializer
+import akka.http.scaladsl.settings.ServerSettings
+import com.typesafe.config.{ Config, ConfigFactory }
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ Await, ExecutionContext, Future }
+import scala.io.StdIn
+import scala.util.{ Failure, Success }
+
+/**
+ * Bootstrap trait. It helps booting up an akka-http by only defining the desired routes.
+ * It offers additional hooks to modify the default behavior
+ */
+trait HttpApp {
```

Comment:
Yes, for sure! I wanted to start with the Scala one and then either make it work for Java or create a dedicated class for it